andela/odin-ah-backend
# #160452490 Add full-text search functionality for articles

#### What does this PR do?
- Adds full-text search functionality for articles created by authors
- Allows user to filter search results by author or tag

#### Description of Task to be completed?
- Write tests
- Create search migration
- Add tsvector column to article table
- Add GIN index to tsvector column
- Add create and update triggers to article table
- Implement pagination of results
- Implement author filter
- Implement tag filter

#### How should this be manually tested?
- Fetch and checkout to the branch
```$xslt
> git fetch
> git checkout ft-160452490-search-functionality
> npm install 
```
- Setup database using .env-sample
* Run Sequelize migration 
```$xslt
 > npm run migration
 ```
- Run application
```$xslt 
> npm run start
```
- To run test
```$xslt 
> npm run test
```

* Create some articles and add tags using the article endpoint. See #24 for details

* Using a REST client such as postman to test endpoints: 
To search for an article: `GET /api/v1/search?q=<query>&limit=<limit>&page=<page>&author=<author>&tag=<tag>` 

* Replace the values in the angle brackets with appropriate inputs. See the API documentation at `http://localhost:3000/api-docs/` for details

* Sample request:
```json
GET /api/v1/search?q=train+dragon&limit=10&page=1&author=3
```

#### Any background context you want to provide?
- None
#### What are the relevant pivotal tracker stories?
- [#160452490](https://www.pivotaltracker.com/story/show/#160452490)
